xarray/tests/test_duck_array_wrapping.py

```diff
@@ -0,0 +1,463 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import xarray as xr
+
+# Don't run cupy in CI because it requires a GPU
+NAMESPACE_ARRAYS = {
+    "jax.numpy": {
+        "attrs": {
+            "array": "ndarray",
+            "constructor": "asarray",
+        },
+        "xfails": {
+            "rolling": "no sliding_window_view",
+            "rolling_mean": "no sliding_window_view",
+        },
+    },
+    "cupy": {
+        "attrs": {
+            "array": "ndarray",
+            "constructor": "asarray",
+        },
+        "xfails": {"quantile": "no nanquantile"},
+    },
+    "pint": {
```

Comment:
Added pint and sparse, and more consolidated xfail lists. Some of these are probably desired behavior, like all the methods that return bool or int not returning a `pint.Quantity`, but it was easiest to just xfail them.